src-plugins/vtkDataMesh/readers/vtkDataMeshReader.cpp

```diff
@@ -89,28 +90,124 @@ bool vtkDataMeshReader::read(const QString& path) {
             return false;
         }
 
-        try
-        {
-            dataSet->Read(path.toLocal8Bit().constData());
```

Comment:
I understand your point below, but it cannot stay that way either, because for instance when you replace the `Read` function here, you consider `if(.vtk)` and `else`. But in `Read` we also manage the mesh format (`ReadMeshFile()`), so we need to find another way